File: src/vovka_pkg/src/find_max_server.cpp
#include "ros/ros.h"
#include "vovka_pkg/FindMax.h"

#include <algorithm>

bool processing(vovka_pkg::FindMax::Request &req,
                vovka_pkg::FindMax::Response & res) {
    
    int maxValue = *std::max_element(req.values.begin(), req.values.end());
    res.max_value = maxValue;
    ROS_INFO("find value: %d", maxValue);
    return true;
}

int main(int argc, char **argv) {
    
    ros::init(argc, argv, "find_max_server");
    ros::NodeHandle node;

    ros::ServiceServer service = node.advertiseService("find_max", processing);
    ROS_INFO("Ready to find max");
    ros::spin();

    return 0;
}

File: src/vovka_pkg/scripts/find_max_server.py
#!/usr/bin/env python

import rospy
from vovka_pkg.srv import *


def processing(req):
    rospy.loginfo('Max value among [%s] is %d' % (str(req.values), max(req.values)))
    return max(req.values)


def find_max_server():
    rospy.init_node('find_max_server', anonymous=True)
    rospy.Service('find_max', FindMax, processing)
    rospy.loginfo('Ready to find max')
    rospy.spin()


if __name__ == '__main__':
    find_max_server()
